src/Profile/Service.php

```diff
@@ -88,7 +88,7 @@ public function getActivityDetails(string $encryptedConnectToken): ActivityDetai
 
         // Check response was successful
         if ($receipt->getSharingOutcome() !== self::OUTCOME_SUCCESS) {
-            throw new ActivityDetailsException('Outcome was unsuccessful', $response);
+            throw new ActivityDetailsException($receipt->generateErrorMessage(), $response, $receipt);
```

Comment:
Seems a bit strange to me to pass a `message` and the `receipt` considering the `message` is actually the result of calling a method on the `receipt`... which then could be done within the `ActivityDetailsException` as it is already doing when calling `receipt->getErrorDetails()`.

Anyway, my 2 cents.